src/lightkurve/io/read.py

```diff
@@ -153,3 +161,39 @@ def read(path_or_url, **kwargs):
             "This file may be corrupt due to an interrupted download. "
             "Please remove it from your disk and try again."
         )
+
+def _read_collection(path_list, products, *, stitch=False, **kwargs):
+    prod_list = []
+    is_lc = (products[0] == TessLightCurve)
+    product_type = "lightcurve" if is_lc else "target pixel file"
+
+    for path in path_list:
+        try:
+            new_prod = read(path, **kwargs)
+
+            if (type(new_prod) in products):
+                prod_list.append(new_prod)
+            else:
+                log.debug(f'Unable to read {path}: The file is not a TESS or Kepler {product_type}.')
+
+        except Exception as e:
+            log.warning(
+                f'Unable to read {path}: {e}. This file will not be added to the collection.'
+            )
+
+    if not prod_list:
+        log.warning(
+            'The resulting collection contains no products.'
+        )
+        return LightCurveCollection([]) if is_lc else TargetPixelFileCollection([])
```

Comment:
can we remove this return statement?  

Since we defined prod_list as empty above, can we not just check if it is empty and print a warning here.  

Then, use the return below?